mosaic2/mosaic2.F90

```diff
@@ -375,34 +375,34 @@ function transfer_to_model_index(istart, iend, refine_ratio)
    return
 
 end function transfer_to_model_index
-!#####################################################################
-function parse_string(string, set, value)
-   character(len=*),  intent(in) :: string
-   character(len=*),  intent(in) :: set
-   character(len=*), intent(out) :: value(:)
-   integer                       :: parse_string
-   integer :: nelem, length, first, last
-
-   nelem = size(value(:))
-   length = len_trim(string)
-
-   first = 1; last = 0
-   parse_string = 0
-
-   do while(first .LE. length)
-      parse_string = parse_string + 1
-      if(parse_string>nelem) then
+
+  function parse_string(string, set, sval)
+  character(len=*),  intent(in) :: string
+  character(len=*),  intent(in) :: set
+  character(len=*), intent(out) :: sval(:)
+  integer                       :: parse_string
+  integer :: nelem, length, first, last
+
+     nelem = size(sval(:))
+     length = len_trim(string)
+
+     first = 1; last = 0
+     parse_string = 0
+
+     do while(first .LE. length)
+       parse_string = parse_string + 1
+       if(parse_string>nelem) then
          call mpp_error(FATAL, "mosaic_mod(parse_string) : number of element is greater than size(value(:))")
-      endif
-      last = first - 1 + scan(string(first:length), set)
-      if(last == first-1 ) then  ! not found, end of string
-         value(parse_string) = string(first:length)
+       endif
+       last = first - 1 + scan(string(first:length), set)
+       if(last == first-1 ) then  ! not found, end of string
```

Comment:
Spacing changes here that need to be reverted for uniformity within the file.

Reply:
should be fixed in 02b33c617615a102c569504e8e0ef3bc5e53f22d

Reply:
@mcallic2 There seems to still be a conflict with main, could you merge it in?